sklearn/utils/estimator_checks.py

```diff
@@ -1147,7 +1151,39 @@ def check_classifiers_train(name, classifier_orig):
                 assert_array_equal(np.argsort(y_log_prob), np.argsort(y_prob))
 
 
-@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_classifiers_multilabel_representation_invariance(name,
+                                                           classifier_orig):
+
+    if name not in MULTI_OUTPUT_CLASSIFIER:
+        raise SkipTest
+
+    X, y = make_multilabel_classification(n_samples=100, n_features=20,
+                                          n_classes=5, n_labels=3,
+                                          length=50, allow_unlabeled=True,
+                                          random_state=0)
+
+    X_train, y_train = X[:80], y[:80]
+    X_test = X[80:]
+
+    y_train_list_of_lists = y_train.tolist()
+    y_train_list_of_arrays = list(y_train)
+
+    classifier = clone(classifier_orig)
+    set_random_state(classifier)
+
+    y_pred = classifier.fit(X_train, y_train).predict(X_test)
+
+    y_pred_list_of_lists = classifier.fit(
+        X_train, y_train_list_of_lists).predict(X_test)
+
+    y_pred_list_of_arrays = classifier.fit(
+        X_train, y_train_list_of_arrays).predict(X_test)
+
+    assert_array_equal(y_pred, y_pred_list_of_arrays)
+    assert_array_equal(y_pred, y_pred_list_of_lists)
```

Comment:
also check that these outputs are, indeed, numeric arrays themselves. (check their type and dtype)